lib/ncr/approval_manager.rb

```diff
@@ -0,0 +1,99 @@
+module Ncr
+  class ApprovalManager
```

Comment:
as long as we are moving stuff around, what about breaking this approval stuff into even smaller bits? This class is certainly more digestible than how it was before, but it's still doing several things.